connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectStubState.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.client
+
+import com.google.protobuf.GeneratedMessageV3
+import io.grpc.stub.StreamObserver
+
+import org.apache.spark.internal.Logging
+
+// This is common state shared between the blocking and non-blocking stubs.
+//
+// The common logic is responsible to verify the integrity of the response. The invariant is
+// that the same stub instance is used for all requests from the same client.
+class SparkConnectStubState extends Logging {
+  // Server side session ID, used to detect if the server side session changed. This is set upon
+  // receiving the first response from the server. This value is used only for executions that
+  // do not use server-side streaming.
+  private var serverSideSessionId: Option[String] = None
+
+  def verifyResponse[RespT <: GeneratedMessageV3](fn: => RespT): RespT = {
+    val response = fn
+    val field = response.getDescriptorForType.findFieldByName("server_side_session_id")
+    // If the field does not exist, we ignore it. New / Old message might not contain it and this
+    // behavior allows us to be compatible.
+    if (field != null) {
+      val value = response.getField(field)
+      serverSideSessionId match {
+        case Some(id) if value != id =>
+          throw new IllegalStateException(s"Server side session ID changed from $id to $value")
```

Comment:
If ReleaseSession returns empty string as server_side_session_id, will it not throw here?